### PR TITLE
Modified bonnie.py to make it run without external disk passage

### PR DESCRIPTION
**Pull request by Naresh-ibm**
* Script was script dependent
* Now it is Independent of disk paasage from yaml
  Providing the disk in ymal file is optional
* Moved the bonnie.py script to io/disk directory as it is a disk
  related test.

Signed-off-by: Naresh Bannoth <nbannoth@in.ibm.com>

### CODE REVIEW

**Comment by Naresh-ibm** [comment]
[bonnie_without_disk.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/845382/bonnie_without_disk.txt)
[bonnie_with_disk.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/845381/bonnie_with_disk.txt)

File: io/disk/bonnie.py [hunk 1]
```diff
@@ -94,8 +95,9 @@ def tearDown(self):
         '''
         Cleanup of disk used to perform this test
         '''
-        self.log.info("Unmounting directory %s", self.scratch_dir)
-        self.part_obj.unmount()
+        if self.disk is not None:
+            self.log.info("Unmounting directory %s", self.scratch_dir)
```
Comment (PraveenPenguin) [inline io/disk/bonnie.py]: ` unmounting disk` message looks good instead of `Unmounting directory`

Comment (nasastry) [inline io/disk/bonnie.py]: how about `"Unmounting disk %s on directory %s", self.disk, self.scratch_dir`

File: io/disk/bonnie.py [hunk 2]
```diff
@@ -64,15 +64,16 @@ def setUp(self):
...
-        self.part_obj.mkfs(self.fstype)
-        self.log.info("Mounting disk %s on directory %s",
-                      self.disk, self.scratch_dir)
-        self.part_obj.mount()
+        if self.disk is not None:
+            self.part_obj = Partition(self.disk, mountpoint=self.scratch_dir)
+            self.log.info("Test will run on %s", self.scratch_dir)
+            self.log.info("Unmounting disk/dir before creating file system")
+            self.part_obj.unmount()
+            self.log.info("creating file system")
```
Comment (nasastry) [inline io/disk/bonnie.py]: Can you please add the kind of file system you are creating?
Example: `"Creating file system %s", fstype`

File: io/disk/bonnie.py (hunk 1 above)
Comment (PraveenPenguin) [inline io/disk/bonnie.py]: yup this looks really good idea

**Comment by Naresh-ibm** [comment]
[Uploading bonnie_without_disk.txt…]()
[Uploading bonnie_with_disk.txt…]()

**Comment by Naresh-ibm** [comment]
amended the suggested changes. please review it and let me know if anything further changes are needed.

**Comment by PraveenPenguin** [comment]
@nasastry thanks

**Pull request merged**